drivers/adodb-mssqlnative.inc.php

```diff
@@ -475,9 +475,12 @@ function _connect($argHostname, $argUsername, $argPassword, $argDatabasename)
 
 		$connectionInfo 			= $this->connectionInfo;
 		$connectionInfo["Database"]	= $argDatabasename;
-		$connectionInfo["UID"]		= $argUsername;
-		$connectionInfo["PWD"]		= $argPassword;
-
+		if (!((is_null($argUsername) || (is_string($argUsername) && strlen($argUsername) === 0)) &&
+			   is_null($argPassword) || (is_string($argPassword) && strlen($argPassword) === 0))) {
```

Comment:
Maybe I'm missing something here, but considering that in PHP `null == ''`, it feels unnecessary to test separately if the parameter is null or empty string; likewise, why the _strlen()_ call and not just compare to empty string ? Just typecast the parameter to string and be done with it...

``` php
if((string)$argUsername != '' || (string)$argPassword != '') {
```

Also, a comment before the if statement explaining why we're doing this would be useful (along the lines of what you wrote in the PR's description).